docs/how-to/incremental-migrations.md

```diff
@@ -0,0 +1,365 @@
+        character.nickname,
+        character.level,
+        character.level + 1,
+    );
+    update_character(
+        ctx,
+        CharacterV2 {
+            level: character.level + 1,
+            ..character
+        },
+    );
+}
+```
+
+And finally, we can define our new `choose_alliance` reducer:
+
+```rust
+#[spacetimedb::reducer]
+fn choose_alliance(ctx: &ReducerContext, alliance: Alliance) {
+    let character = find_character_for_player(ctx);
+    log::info!(
+        "Setting alliance of {} to {:?}",
+        character.nickname,
+        alliance,
+    );
+    update_character(
+        ctx,
+        CharacterV2 {
+            alliance,
+            ..character
+        },
+    );
+}
+```
+
+A bit more playing around with the CLI will show us that everything works as intended:
+
+```sh
+# Our row in `character` still exists:
+$ spacetime sql incr-migration-demo 'SELECT * FROM character'
+
+ player_id | nickname | level | class
+-----------+----------+-------+----------------
+ <snip>    | "Gefjon" | 2     | (Fighter = ())
+
+# We haven't triggered the "Gefjon" row to migrate yet, so `character_v2` is empty:
+$ spacetime sql -s local incr-migration-demo 'SELECT * FROM character_v2'
+
+ player_id | nickname | level | class | alliance
+-----------+----------+-------+-------+----------
+
+# Accessing our character, e.g. by leveling up, will cause it to migrate into `character_v2`:
+$ spacetime call incr-migration-demo level_up_character
+
+2025-01-07T16:00:20.500600Z  INFO: src/lib.rs:110: Leveling up Gefjon from 2 to 3
+
+# Now `character_v2` is populated:
+$ spacetime sql incr-migration-demo 'SELECT * FROM character_v2'
+
+ player_id | nickname | level | class          | alliance
+-----------+----------+-------+----------------+----------------
+ <snip>    | "Gefjon" | 3     | (Fighter = ()) | (Neutral = ()) 
+
+# The original row in `character` still got updated by `level_up_character`,
+# so outdated clients can continue to function:
+$ spacetime sql incr-migration-demo 'SELECT * FROM character'
+
+ player_id | nickname | level | class
+-----------+----------+-------+----------------
+ <snip>    | "Gefjon" | 3     | (Fighter = ())
+
+# We can set our alliance:
+$ spacetime call incr-migration-demo choose_alliance '{ "Good": {} }'
+
+2025-01-07T16:13:53.816501Z  INFO: src/lib.rs:129: Setting alliance of Gefjon to Good
+
+# And that change shows up in `character_v2`:
+$ spacetime sql incr-migration-demo 'SELECT * FROM character_v2'
+
+ player_id | nickname | level | class          | alliance
+-----------+----------+-------+----------------+-------------
+ <snip>    | "Gefjon" | 3     | (Fighter = ()) | (Good = ())
+
+# But `character` is not changed, since it doesn't know about alliances:
+$ spacetime sql incr-migration-demo 'SELECT * FROM character'
+
+ player_id | nickname | level | class
+-----------+----------+-------+----------------
+ <snip>    | "Gefjon" | 3     | (Fighter = ())
+```
+
```

Comment:
To maintain something like this long-term, we'd like recurring checks that data is synced up across versions. I would recommend a repeating reducer that runs hourly, say, that does this check.

It would be even better to write a check constraint on the join between the data tables, but we don't support that yet.